android/buildAndroidBOINC-CI.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/sh
+set -e
+
+# This file is part of BOINC.
+# http://boinc.berkeley.edu
+# Copyright (C) 2018 University of California
+#
+# BOINC is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License
+# as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# BOINC is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+export ANDROID_HOME=$HOME/Android/Sdk
+export NDK_ROOT=$HOME/Android/Ndk
+export ANDROID_TC=$HOME/Android/Toolchains
+
+# checks if a given path is canonical (absolute and does not contain relative links)
+# from http://unix.stackexchange.com/a/256437
+isPathCanonical() {
+  case "x$1" in
+    (x*/..|x*/../*|x../*|x*/.|x*/./*|x./*)
+        rc=1
+        ;;
+    (x/*)
+        rc=0
+        ;;
+    (*)
+        rc=1
+        ;;
+  esac
+  return $rc
+}
+
+doclean=""
+cache_dir=""
+arch=""
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --cache_dir)
+        cache_dir="$2"
+        shift
+        ;;
+        --clean)
+        doclean="yes"
+        ;;
+        --arch)
+        arch="$2"
+        shift
+        ;;
+        *)
+        echo "unrecognized option $key"
+        ;;
+    esac
+    shift # past argument or value
+done
+
+if [ "x$cache_dir" != "x" ]; then
+    if isPathCanonical "$cache_dir" && [ "$cache_dir" != "/" ]; then
+         PREFIX="$cache_dir"
+     else
+         echo "cache_dir must be an absolute path without ./ or ../ in it"
+         exit 1
+     fi
+else
+    cd ../
+    PREFIX="$(pwd)/3rdParty/buildCache/android"
+    cd android/
+fi
+
+mkdir -p "${PREFIX}"
```

Comment:
You actually need two directories at this point. One where you are going to download the source and build the dependencies (`buildDir`) and one where you are going to store the build products of those dependencies (`cacheDir`). Right now you are using the same directory for both which bloats the cache. 

At the end of this script you need to set `ANDROID_TC` to be `cacheDir` so the other scripts following can install the libraries into the right directory. For the `buildDir` you need to have another directory specified. For Linux I hardcoded this to `3rdParty/linux`. Since you are downloading the sources here and export the build directory via a variable you can do the same or you could add another argument to the script so the `buildDir` can be specified.

I would set the variables like this for the default case:
```
BUILD_DIR="$(pwd)/3rdParty/android"
PREFIX="$(pwd)/3rdParty/buildCache/android-tc
```
Note that `PREFIX` now has the same meaning as in the Linux and Mac build scripts. Following comments will refer to those two variables.

Reply:
Done. Please make one more review. Thanks